manifests/prometheus.yml

```diff
@@ -0,0 +1,547 @@
+  default:
+    azs: [z1]
+  prometheus_port:   (( grab params.prometheus_port   || 8080 ))
+  grafana_port:      (( grab param.grafana_port       || 443 ))
+  alertmanager_port: (( grab params.alertmanager_port || 8082 ))
+
+exodus:
+  prometheus_url:   (( concat params.external_domain ":" meta.prometheus_port    ))
+  grafana_url:      (( concat params.external_domain ":" meta.grafana_port       ))
+  alertmanager_url: (( concat params.external_domain ":" meta.alertmanager_port  ))
+
+instance_groups:
+- name:      prometheus
+  instances: 1
+  azs:       (( grab params.availability_zones || meta.default.azs ))
+  stemcell:  default
+  persistent_disk_type: (( grab params.persistent_disk_type || "prometheus" ))
+  networks:   
+    - name: (( grab params.network || "default" ))
+      static_ips: (( grab params.static_ip ))
+  vm_type:   (( grab params.vm_type || "default" ))
+  jobs:
+  - name: alertmanager
+    properties:
+      alertmanager:
+        mesh:
+          password: (( vault meta.vault "/alertmanager:password" ))
+        receivers:
+        - name: default
+        route:
+          receiver: default
+        test_alert:
+          daily: true
+    release: prometheus
+  - name: prometheus2
+    properties:
+      prometheus:
+        rule_files:
+        - /var/vcap/jobs/postgres_alerts/*.alerts.yml
+        - /var/vcap/jobs/prometheus_alerts/*.alerts.yml
+        - /var/vcap/jobs/bosh_alerts/*.alerts.yml
+        scrape_configs:
+        - file_sd_configs:
+          - files:
+            - /var/vcap/store/bosh_exporter/bosh_target_groups.json
+          job_name: prometheus
+          relabel_configs:
+          - action: keep
+            regex: prometheus\d?
+            source_labels:
+            - __meta_bosh_job_process_name
+          - regex: (.*)
+            replacement: ${1}:9001
+            source_labels:
+            - __address__
+            target_label: __address__
+        - job_name: bosh
+          scrape_interval: 2m
+          scrape_timeout: 1m
+          static_configs:
+          - targets:
+            - localhost:9190
+        - file_sd_configs:
+          - files:
+            - /var/vcap/store/bosh_exporter/bosh_target_groups.json
+          job_name: bosh_tsdb
+          relabel_configs:
+          - action: keep
+            regex: bosh_tsdb_exporter
+            source_labels:
+            - __meta_bosh_job_process_name
+          - regex: (.*)
+            replacement: ${1}:9194
+            source_labels:
+            - __address__
+            target_label: __address__
+        - file_sd_configs:
+          - files:
+            - /var/vcap/store/bosh_exporter/bosh_target_groups.json
+          job_name: cadvisor
+          relabel_configs:
+          - action: keep
+            regex: cadvisor
+            source_labels:
+            - __meta_bosh_job_process_name
+          - regex: (.*)
+            replacement: ${1}:8080
+            source_labels:
+            - __address__
```

Comment:
Also, while we're cleaning up this YAML for readability, single-item lists work great with YAML flow notation:

```
source_labels: [__address__]
```